Bencodex/Types/Dictionary.cs

```diff
@@ -108,26 +342,34 @@ public bool TryGetKey(IKey equalKey, out IKey actualKey)
             return Value.TryGetKey(equalKey, out actualKey);
         }
 
+        public T GetValue<T>(string name)
+            where T : IValue
+        {
+            return (T)this[name];
+        }
+
+        public static Dictionary Empty => new Dictionary();
+
         public override bool Equals(object obj)
         {
             switch (obj)
             {
                 case null:
                     return false;
-                case IImmutableDictionary<IKey, IValue> d:
+                case Dictionary d:
                     return (
-                        (IEquatable<IImmutableDictionary<IKey, IValue>>) this
+                        (IEquatable<Dictionary>) this
                     ).Equals(d);
                 default:
                     return false;
             }
         }
 
-        bool IEquatable<IImmutableDictionary<IKey, IValue>>.Equals(
-            IImmutableDictionary<IKey, IValue> other
+        bool IEquatable<Dictionary>.Equals(
+            Dictionary other
         )
         {
-            if (this.LongCount() != other.LongCount()) return false;
+            if (Value.LongCount() != other.Value.LongCount()) return false;
```

Comment:
Is this change necessary?

Reply:
It seems there are lines remained although `Dictionary` became to implement `IImmutableDictionary<IKey, IValue>` again. I will see the lines again and apply your reviews.